### PR TITLE
prov/verbs: Remove unneeded mr_mode bits

### PR DESCRIPTION
**Pull request by gladkovdmitry17**
Signed-off-by: Dmitry Gladkov <dmitry.gladkov@intel.com>

### CODE REVIEW

File: prov/verbs/src/verbs_info.c [hunk 1]
```diff
@@ -84,7 +84,7 @@ const struct fi_domain_attr verbs_domain_attr = {
 	.control_progress	= FI_PROGRESS_AUTO,
 	.data_progress		= FI_PROGRESS_AUTO,
 	.resource_mgmt		= FI_RM_ENABLED,
-	.mr_mode		= OFI_MR_BASIC_MAP | FI_MR_LOCAL | FI_MR_BASIC,
+	.mr_mode		= FI_MR_LOCAL | FI_MR_BASIC,
```
Comment (shefty) [inline prov/verbs/src/verbs_info.c]: I think verbs actually needs MR_BASIC_MAP

Comment (gladkovdmitry17) [reply prov/verbs/src/verbs_info.c]: Do you think this is needed for internal purposes? I don't see any code dependencies on OFI_MR_BASIC_MAP

Comment (a-ilango) [reply prov/verbs/src/verbs_info.c]: Verbs needs apps to support OFI_MR_BASIC_MAP bits.
> I don't see any code dependencies on OFI_MR_BASIC_MAP

I don't think there is any code in the provider to handle an app not supporting OFI_MR_BASIC_MAP bits.

Comment (shefty) [reply prov/verbs/src/verbs_info.c]: @gladkovdmitry17 - verbs devices require the use of virtual addressing, provider allocated keys, and that memory regions be backed by physical pages, which is what MR_BASIC_MAP indicates.

Comment (shefty) [reply prov/verbs/src/verbs_info.c]: Actually, if the provider sets FI_MR_BASIC it means that it supports that mode, not that it requires it.  This is needed for backwards compatibility and is used by the socket provider to support apps that are only coded for FI_MR_BASIC.

For >= 1.5, the provider just sets the mr_mode bits that they actually require.  If the app sets individual mr_mode bits, then the provider can clear any of them.  If the app specifies FI_MR_BASIC, the provider must support that exact semantic (which is equivalent to keeping all MR_BASIC_MAP bits set).

**Pull request closed without merging**